### ISSUE TITLE
ClassAndModuleChildren compact syntax false negative when outer module is already nested

### ISSUE THREAD

**Issue by mockdeep** [rubocop/rubocop#9761]
When using the `ClassAndModuleChildren` rule with `EnforcedStyle: compact`, it does not report violations when the outer module is already nested:

```rb
module Foo::Bar
  class Baz
  end
end
```

--------

## Expected behavior

The above would report a violation. It should require:

```rb
class Foo::Bar::Baz
end
```

## Actual behavior

It does not report a violation.

## Steps to reproduce the problem

See above.

## RuboCop version

```
1.13.0 (using Parser 3.0.1.0, rubocop-ast 1.4.1, running on ruby 3.0.1 x86_64-linux)
  - rubocop-rails 2.9.1
  - rubocop-rspec 2.3.0
```
